### PR TITLE
Design decision template

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

**Comment by bmmoore** [comment]
The format looks nice, but I think there should also be a template that's not already filled out with example text.

Your suggestion for obsolete files is unclear - is adding an `OBSOLETE` prefix supposed to override the earlier decision that filenames should start with the date? If so, that should be mentioned, if not then the word "prefixed" is not an adequate description of how the filename should be modified.

File: design-decisions/0000-00-00-Template.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+This is not supposed to be a burden to write or read, 1-2 paragraphs should be
+enough.
+
+If the decision changes the way we do something, please mention this here and
+describe the current way either here or in the "Other options" section.
+
+Example:
+
+    We are writing decision documents for each decision that we want to record.
+    These live in the `desing-decisions` directory, using file names with
```
Comment (traiansf) [inline design-decisions/0000-00-00-Template.md]: `desing-decisions` => `design-decisions`

**Comment by virgil-serbanuta** [comment]
@bmmoore I split this in two parts, the template itself, and the decision to have a template. I still found it useful to leave some text inside the template. Does that seem reasonable?

File: design-decisions/2018-09-20-Decision-Template.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Background
+----------
+
+We are writing decision documents for each decision that we want to record.
+These live in the `design-decisions` directory, using file names with
+the `yyyy-mm-dd-<dash-separated-title>.md` format, with the date
+being roughly the date when the decision was made.
+
+Currently (2018-09-19) the document format is unrestricted, but loosely
```
Comment (ttuegel) [inline design-decisions/2018-09-20-Decision-Template.md]: Now that this file has a date in the name, we probably don't need to specify the date in the text.

**Pull request merged**